drivers/gpu/drm/panel/panel-simple.c

```diff
@@ -4681,23 +4685,141 @@ static const struct of_device_id dsi_of_match[] = {
 	}, {
 		.compatible = "osddisplays,osd101t2045-53ts",
 		.data = &osd101t2045_53ts
+	}, {
+		/* Must be the last entry */
+		.compatible = "panel-dsi",
+		.data = &panel_dsi,
 	}, {
 		/* sentinel */
 	}
 };
 MODULE_DEVICE_TABLE(of, dsi_of_match);
 
+
+/* Checks for DSI panel definition in device-tree, analog to panel_dpi */
+static int panel_dsi_dt_probe(struct device *dev,
+			  struct panel_desc_dsi *desc_dsi)
+{
+	struct panel_desc *desc;
+	struct display_timing *timing;
+	const struct device_node *np;
+	const char *dsi_color_format;
+	const char *dsi_mode_flags;
+	struct property *prop;
+	int dsi_lanes, ret;
+
+	np = dev->of_node;
+
+	desc = devm_kzalloc(dev, sizeof(*desc), GFP_KERNEL);
+	if (!desc)
+		return -ENOMEM;
+
+	timing = devm_kzalloc(dev, sizeof(*timing), GFP_KERNEL);
+	if (!timing)
+		return -ENOMEM;
+
+	ret = of_get_display_timing(np, "panel-timing", timing);
+	if (ret < 0) {
+		dev_err(dev, "%pOF: no panel-timing node found for \"panel-dsi\" binding\n",
+			np);
+		return ret;
+	}
+
+	desc->timings = timing;
+	desc->num_timings = 1;
+
+	of_property_read_u32(np, "width-mm", &desc->size.width);
+	of_property_read_u32(np, "height-mm", &desc->size.height);
+
+	dsi_lanes = drm_of_get_data_lanes_count(np, 1, 4);
```

Comment:
Prototype of drm_of_get_data_lanes_count is 
```
int drm_of_get_data_lanes_count(const struct device_node *endpoint,
				const unsigned int min, const unsigned int max)
```
You've passed the device_node of the base device, not the endpoint.

Use ```drm_of_get_data_lanes_count_ep(np, 0, -1, 1, 4);```

Reply:
the property does not sit in the endpoint though but is part of the panel-dsi binding to be able to pass that into the dsi_mode for setup.
`drm_of_get_data_lanes_count_ep` would not be able to find the property then. If you check the implementation of the functions the use of `endpoint` of the non `_ep` prototype for it is a bit misleading because it does not actually expect an endpoint it just looks for the data-lanes property within the node it was given and does a boundary check, nothing specific to endpoints in there.
`drm_of_get_data_lanes_count_ep` does look for an endpoint though and expects to find the property there.
So I think in this case `drm_of_get_data_lanes_count` is more apropriate. Otherwise I would not know how to specify the binding properly without modifying the inherited bindings.

Reply:
switched data-lanes to be in endpoint